src/qz/utils/PrintingUtilities.java

```diff
@@ -59,16 +82,7 @@ public static PrinterResolution getNativeDensity(PrintService service) {
         PrinterResolution pRes = (PrinterResolution)service.getDefaultAttributeValue(PrinterResolution.class);
 
         if (pRes == null && !SystemUtilities.isWindows()) {
-            if (CUPS_DPI == null || !CUPS_DESC.containsValue(service.getName())) {
-                CUPS_DESC = ShellUtilities.getCupsPrinters();
-                CUPS_DPI = ShellUtilities.getCupsDensities(CUPS_DESC);
-            }
-
-            for(String name : CUPS_DESC.keySet()) {
-                if (CUPS_DESC.get(name).equals(service.getName())) {
-                    return CUPS_DPI.get(name);
-                }
-            }
+            return CUPS_DPI.get(getPrinterId(service.getName()));
```

Comment:
I realize this is confusing (calling a `.get` on a variable that was never initialized) but should be perfectly valid from an order of operations perspective.